vaadin-date-time-picker-flow-parent/vaadin-date-time-picker-flow/src/main/java/com/vaadin/flow/component/datetimepicker/DateTimePicker.java

```diff
@@ -416,97 +420,67 @@ public Optional<String> getAriaLabel() {
     }
 
     /**
-     * Contains DateTimePicker internalization properties
-     */
-    public static class DateTimePickerI18n implements Serializable {
-        private String dateLabel;
-        private String timeLabel;
-
-        public String getDateLabel() {
-            return dateLabel;
-        }
-
-        public DateTimePickerI18n setDateLabel(String dateLabel) {
-            this.dateLabel = dateLabel;
-            return this;
-        }
-
-        public String getTimeLabel() {
-            return timeLabel;
-        }
-
-        public DateTimePickerI18n setTimeLabel(String timeLabel) {
-            this.timeLabel = timeLabel;
-            return this;
-        }
-    }
-
-    /**
-     * Sets the accessible label for the date picker.
+     * Sets the aria-label suffix for the date picker.
+     * <p>
+     * The suffix set with this method takes precedence over the suffix set with
+     * {@link DateTimePickerI18n#setDateLabel(String)}.
      * <p>
-     * The final value is a concatenation of the accessible label from
-     * DateTimePicker's {@link #getAriaLabel()} or {@link #getLabel()} and the
-     * given accessible label.
+     * The date picker's final aria-label is a concatenation of DateTimePicker's
+     * {@link #getAriaLabel()} or {@link #getLabel()} and this suffix.
      *
      * @param dateLabel
-     *            the value to be used as part of date picker aria-label.
+     *            the value to be used as a suffix in the date picker
+     *            aria-label.
      */
     public void setDateAriaLabel(String dateLabel) {
-        if (i18n == null) {
-            i18n = new DateTimePickerI18n();
-        }
-        i18n.setDateLabel(dateLabel);
-        getElement().setPropertyJson("i18n", JsonSerializer.toJson(i18n));
+        dateAriaLabel = dateLabel;
+        updateI18n();
     }
 
     /**
-     * Gets the accessible label of the date picker.
+     * Gets the aria-label suffix for the date picker.
      * <p>
-     * Note that this method will return the last value passed to
+     * Note: this method will return the last value passed to
      * {@link #setDateAriaLabel(String)}, not the value currently set on the
      * `aria-label` attribute of the date picker input element.
      *
      * @return an optional label or an empty optional if no label has been set
+     *         with this method before.
      */
     public Optional<String> getDateAriaLabel() {
-        if (i18n == null) {
-            return Optional.empty();
-        }
-        return Optional.ofNullable(i18n.getDateLabel());
+        return Optional.ofNullable(dateAriaLabel);
```

Comment:
Do we consider this a minor breaking change here?

Reply:
Not sure, could you explain what the breaking change is here?

Reply:
Ah nevermind, since the i18n instance was not accessible before this can not result in a change in behavior.